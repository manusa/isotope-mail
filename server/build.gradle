buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
