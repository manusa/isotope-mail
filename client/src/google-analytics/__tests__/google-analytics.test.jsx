import React from 'react';
import {shallow} from 'enzyme/build';

describe('Google Analytics component test suite', () => {
  let googleAnalyticsTrackingId;
  let GoogleAnalytics;
  beforeEach(() => {
    jest.mock('../selectors');
    googleAnalyticsTrackingId = require('../selectors').googleAnalyticsTrackingId;
    GoogleAnalytics = require('../').default;
  });
  describe('Snapshot render', () => {
    test('Tracking ID set in global variable, should render script', () => {
      // Given
      googleAnalyticsTrackingId.mockImplementationOnce(() => 'UA-1337-33');
      // When
      const routes = shallow(<GoogleAnalytics />);
      // Then
      expect(routes).toMatchSnapshot();
    });
    test('Tracking ID NOT set in global variable, should render empty', () => {
      // Given
      googleAnalyticsTrackingId.mockImplementationOnce(() => null);
      // When
      const routes = shallow(<GoogleAnalytics />);
      // Then
      expect(routes).toMatchSnapshot();
    });
  });
});
