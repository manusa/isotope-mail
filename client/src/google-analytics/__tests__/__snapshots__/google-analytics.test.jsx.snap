// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Analytics component test suite Snapshot render Tracking ID NOT set in global variable, should render empty 1`] = `""`;

exports[`Google Analytics component test suite Snapshot render Tracking ID set in global variable, should render script 1`] = `
<Fragment>
  <script
    async={true}
    src="https://www.googletagmanager.com/gtag/js?id=UA-1337-33"
  />
  <script>
    
window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-1337-33');
      
  </script>
</Fragment>
`;
